swarm_node.py

```diff
@@ -149,6 +154,9 @@ def send_transfer(tag, messages, address, values, dict_tips, debug=0):
         time_end_pow = time.time()
         elapsed_pow = elapsed_pow + (time_end_pow - time_start_pow)
 
+        # Update previous tx hash for next transaction
+        prev_tx = Transaction.from_tryte_string(tx_tryte[0:2673]).hash
```

Comment:
Is it possible that `prev_tx` is set to null?

Reply:
It wouldn't be null as long as `tx_tryte`  has valid trytes. Even if trytes are invalid, it should yield errors when building bundle already.

Reply:
So, the check of `prev_tx` is only valid for the first time.

Reply:
Sorry, I still don't get the problem here. Trytes mentioned above are each transaction trytes instead of `trytes` in code which are whole bundle trytes. The possible error I could come up with is `tx_tryte` is wrong after PoW , but that would just get incorrect transaction hash.

Reply:
It is a matter of premature optimizations where branches are eliminated. I just wanted to get rid of the check for `prev_tx` for first intention.